core/includes/file.inc

```diff
@@ -1107,16 +1150,37 @@ function file_move(File $source, $destination = NULL, $replace = FILE_EXISTS_REN
  *   - FILE_EXISTS_ERROR - Do nothing and return FALSE.
  *
  * @return
- *   The URI of the moved file, or FALSE in the event of an error.
+ *   The path to the new file, or FALSE in the event of an error.
  *
  * @see file_move()
  */
 function file_unmanaged_move($source, $destination = NULL, $replace = FILE_EXISTS_RENAME) {
-  $filepath = file_unmanaged_copy($source, $destination, $replace);
-  if ($filepath == FALSE || file_unmanaged_delete($source) == FALSE) {
+  if (!file_unmanaged_prepare($source, $destination, $replace)) {
     return FALSE;
   }
-  return $filepath;
+  // Ensure compatibility with Windows.
+  // @see backdrop_unlink()
+  if ((substr(PHP_OS, 0, 3) == 'WIN') && (!file_stream_wrapper_valid_scheme(file_uri_scheme($source)))) {
+    chmod($source, 0600);
+  }
```

Comment:
This entire bit is confusing. I noted that David_Rothstein said the same thing in his code review on Drupal. It's not clear why this would affect Windows only.

In order to move a file, write permissions are necessary both on the source and the destination, since the file is going to be moved from the previous location. Since we use `backdrop_chmod($destination)` on the destination, the moved file is going to have the standard Backdrop permissions. I think if we want to avoid permissions problems, we should simply do the same thing on `$source` prior to the move. It would accomplish the same thing to ensure the file is writable for Windows computers, if they actually have a problem in the first place.

So I would suggest the following:

```
  // The file must be writable in the old location before moving.
  backdrop_chmod($source);
```

Reply:
Agreed. Done.